pkg/controller/migmigration/task.go

```diff
@@ -202,31 +220,22 @@ func (t *Task) Run() error {
 			log.Trace(err)
 			return err
 		}
-		t.Phase = EnsureStagePods
+		t.next()
 	case EnsureStagePods:
-		count, err := t.ensureStagePodsCreated()
+		_, err := t.ensureStagePodsCreated()
 		if err != nil {
 			log.Trace(err)
 			return err
 		}
-		if count > 0 {
```

Comment:
The other change is we're no iterating over StagePodsCreated, RestartRestic, and ResticRestarted even if there are no stage pods. As long as those operations are still successful when there are no stage pods, this should be fine. The prior logic skipped these steps if there were no stage pods.

Reply:
As you know, the current logic generates the PVs list based on application pods that are bound to PVs.  And for each of those (app) pods, we create a stage pod.  So, if there are PVs, there will be stage pods.  Going forward, if we find a way to include PVs not assocated with pods, we can change the logic to count the stage pods.

For now, I think it's safe to use the PVS flag to qualify the phases as you indicated.

Reply:
So if every PV is being copied by snapshot rather than by restic we may not have any stage pods.

Reply:
Ah.  Okay.

Reply:
Fixed.